src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```
@@ -797,4 +850,45 @@ void checkQueryOptionsSupportability(QueryOptions queryOptions) {
             throw new FeatureNotSupportedException("TODO-HIBERNATE-70 https://jira.mongodb.org/browse/HIBERNATE-70");
         }
     }
+
+    private static AstComparisonFilterOperator getAstComparisonFilterOperator(ComparisonOperator operator) {
+        return switch (operator) {
+            case EQUAL -> EQ;
+            case NOT_EQUAL -> NE;
+            case LESS_THAN -> LT;
+            case LESS_THAN_OR_EQUAL -> LTE;
+            case GREATER_THAN -> GT;
+            case GREATER_THAN_OR_EQUAL -> GTE;
+            default -> throw new FeatureNotSupportedException("Unsupported comparison operator: " + operator.name());
+        };
+    }
+
+    private static boolean visitFieldPath(Expression expression) {
+        return expression instanceof ColumnReference || expression instanceof BasicValuedPathInterpretation;
+    }
+
+    private static BsonValue toBsonValue(@Nullable Object queryLiteral) {
+        if (queryLiteral == null) {
+            return BsonNull.VALUE;
```

Currently, FROM `Contact WHERE age = null` is translated to `{ "age": { "$eq": null } }`, which differs from what a user would expect using HQL/SQL might expect - as such a condition would omit results. Is this expected behaviour?

I didn’t find a test covering this scenario, so I assume handling null comparisons may be out of scope for this PR. if yes, should we throw an unsupported exception for now, with a reference to a follow-up JIRA ticket?

---

Yeah, null or missing field is out of scope of this PR and a separate ticket was created for that purpose: https://jira.mongodb.org/browse/HIBERNATE-74

---

I added logic to throw exception when null literal value is used.

As Jeff reminded, we should add similar safety net against parameter usage with null value. I added a new `checkJdbcParameterBindings` method to throw exception when any of the parameters is using null value.

https://github.com/mongodb/mongo-hibernate/pull/81/commits/951e64d24f7ec8ccc7e14a07103900196896fe72